### DISCUSSION TITLE
merging overlapping objects

### DISCUSSION THREAD

**Discussion by danafd** [marcomusy/vedo#1010] [General]
Hi, 

If I merge 2 objects that partially overlap using the `merge()` function, I noticed that the output of `volume()` is the sum of their individual volumes and not the volume of the merged entity. Is there a way to measure the volume of the newly formed single entity? 

What i'm aiming to do: 

I'm trying to measure the volume of the proportion of a mesh that is embedded within another mesh but using `cut_with_mesh()` creates non-manifold, non-closed meshes preventing me from measuring the precise volume of the embedded region. So, I'm considering merging both meshes, measuring the volume of the new entity, then subtracting the volumes of the larger mesh to get the volume of the protruding mesh and finally the volume of the embedded mesh. 

Thanks for the help!

**Comment by marcomusy** [comment]
If the meshes are noisy this might become a difficult problem... You can try with boolean operations eg:
```py
from vedo import *

cube1 = Cube().triangulate().color('r5')
cube2 = Cube().triangulate().color('g5').scale(0.99).shift(.25,0,0)

bi = cube1.boolean('intersect', cube2).alpha(0.5)

cube1.wireframe()
cube2.wireframe()

show(cube1, cube2, bi, f"Volume = {bi.volume()}", axes=1)
```

![Screenshot from 2024-01-10 17-06-16](https://github.com/marcomusy/vedo/assets/32848391/63028793-c72a-4b67-a1c3-001234eee886)

Also check out `examples/other/remesh_meshfix.py` to fix problematic meshes,
and `examples/other/remesh_ACVD.py` to possibly downsample them before the boolean operation.

**Reply by danafd** [reply]
I'll try it out. thanks for the help!

**Comment by danafd** [comment]
Hello again, 
I come back to this problem after a while. The proposed methods don't seem to work. 

However, I tried to binarize the merged mesh using `mergedmesh.binarize()`, turn the binarised volume into a numpy array using the `tonumpy()` function and then it back into a mesh. The `newmesh.volume()` result makes sense now since i got rid of the overlap between meshes. 

But I noticed that the binarize() function creates a volume with different dimesions than the original mesh. I read the mesh from a binary numpy array of shape (66, 1024, 1024) and the binarised volume becomes completely different. I would like my newmesh to have the same dimesions as the original mesh for further analyses. Would this be possible with binarize()?

Apologies if this sounds confusing, I've been trying to find creative ways to solve this issue. 

Thanks for the help!

**Reply by marcomusy** [reply]
Hi, yes- you need to specify the spacing (a numpy array cannot contain this information!).
the spacing is the size in x,y,z of a single voxel.

```
vol.spacing([...])
```
